internal/services/paloalto/local_rule_stack_rule_resource.go

```diff
@@ -0,0 +1,542 @@
+				props.ProtocolPortList = pointer.To(model.ProtocolPorts)
+			}
+
+			if model.Protocol != "" && !strings.EqualFold(model.Protocol, protocolApplicationDefault) && len(model.ProtocolPorts) == 0 {
+				props.Protocol = pointer.To(model.Protocol)
+			}
+
+			if _, err = client.CreateOrUpdate(ctx, id, localrules.LocalRulesResource{Properties: props}); err != nil {
+				return err
+			}
+
+			metadata.SetID(id)
+
+			if err = rulestackClient.CommitThenPoll(ctx, *rulestackId); err != nil {
+				return fmt.Errorf("committing Local Rulestack config for %s: %+v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r LocalRuleStackRule) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.PaloAlto.LocalRulesClient
+
+			id, err := localrules.ParseLocalRuleID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var state LocalRuleModel
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(existing.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("reading %s: %+v", *id, err)
+			}
+
+			state.RuleStackID = localrulestacks.NewLocalRulestackID(id.SubscriptionId, id.ResourceGroupName, id.LocalRulestackName).ID()
+			p, err := strconv.Atoi(id.LocalRuleName)
+			if err != nil {
+				return fmt.Errorf("parsing Rule Priortiy for %s: %+v", *id, err)
+			}
+			state.Priority = p
+
+			props := existing.Model.Properties
+			state.Name = props.RuleName
+			state.Action = string(pointer.From(props.ActionType))
+			state.Applications = pointer.From(props.Applications)
+			state.AuditComment = pointer.From(props.AuditComment)
+			state.Category = schema.FlattenCategory(props.Category)
+			state.DecryptionRuleType = string(pointer.From(props.DecryptionRuleType))
+			state.Description = pointer.From(props.Description)
+			state.Destination = schema.FlattenDestination(props.Destination)
+			state.LoggingEnabled = stateEnumAsBool(props.EnableLogging)
+			if certName := pointer.From(props.InboundInspectionCertificate); certName != "" {
+				state.InspectionCertificateID = certificates.NewLocalRulestackCertificateID(id.SubscriptionId, id.ResourceGroupName, id.LocalRulestackName, certName).ID()
+			} else {
+				state.InspectionCertificateID = certName
+			}
+			state.NegateDestination = boolEnumAsBoolRule(props.NegateDestination)
+			state.NegateSource = boolEnumAsBoolRule(props.NegateSource)
+			if v := pointer.From(props.Protocol); !strings.EqualFold(v, protocolApplicationDefault) {
+				state.Protocol = pointer.From(props.Protocol)
+			} else {
+				state.Protocol = protocolApplicationDefault
+			}
+			state.ProtocolPorts = pointer.From(props.ProtocolPortList)
+			state.RuleEnabled = stateEnumAsBool(props.RuleState)
+			state.Source = schema.FlattenSource(props.Source)
+			state.Tags = flattenTagsFromRule(props.Tags)
```

Comment:
as in other cases, can we conditionally set these only when we have a `model` object?